pkg/config/config.go

```diff
@@ -37,37 +37,37 @@ var (
 )
 
 type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+	// URL string `json:"url"`
```

Comment:
@benluddy Any comments on this?

Reply:
> but is instead used to specify the IP and port this instance's API server binds to.

I don't see why we would want that to be configurable, the address the API server binds to should always be the same.

The url of the API server will also be present in the kubeconfig so other instances can use the kubeconfig to connect to the server.

Reply:
We can't assume that network addresses stay the same. MicroShift hosts can be autonomous devices that move between networks.

Hosts may have multiple NICs and multiple IPs.

Given those constraints, how do we ensure that the correct API service IP is used?

Reply:
Then we need dns if the ips are dynamic and we can't trust them. Adding a route to reach out to the API server doesn't sound too bad if we need to support scenarios where hosts have multiple NICs and IPs and they still need an immutable way to reach out to the API server.

Also, can you confirm that these constraints are coming from the initial customers of microshift? We are trying to come up with a minimal version of the config to avoid having options configurable that we don't want to support in the future but can't remove because some of the initial customers may already depend on them.